cmake/toolchain/cross-compile/generic.cmake

```diff
@@ -29,3 +29,5 @@ set(LINKER ld)
 set(BINTOOLS gnu)
 
 message(STATUS "Found toolchain: cross-compile (${CROSS_COMPILE})")
+
+set(TOOLCHAIN_HAS_NEWLIB ON CACHE BOOL "True if toolchain supports newlib")
```

Comment:
no, when using `cross-compile` we really cannot know if the toolchain supports newlib or not, this information must come from the user.

If using `ZEPHYR_TOOLCHAIN_VARIANT=cross-compile` and you know the toolchain supports newlib, then you may specify this during build using: `-DTOOLCHAIN_HAS_NEWLIB=y`.

**Note**: you may propose that cross compile supports fetching the `TOOLCHAIN_HAS_HEWLIB` (perhaps prefixed with `CROSS_COMPILE`) from env in addition to CMake cache, as this is probably a value which is quite static.

Reply:
Thanks for your comments, I understand we need to pass the information for toolchain support newlib to Zephyr, but I think it is not very easy to use: -DTOOLCHAIN_HAS_NEWLIB=y, we need to update cmake command or script. how about change TOOLCHAIN_HAS_NEWLIB to be a configurable item in Zephyr, such as CONFIG_TOOLCHAIN_HAS_NEWLIB, so that the application can enable it if toolchain supports newlib.
If this is not fine, as currently there is no mechanism to detect whether toolchain supports newlib or not automatically, at least we need to report some error message in case of application want to enalbe CONFIG_NEWLIB_LIBC, but TOOLCHAIN_HAS_NEWLIB=n.

Reply:
> so that the application can enable it if toolchain supports newlib.

but an application can be built with many different toolchains, so how can an app know whether or not a given toolchain supports newlib or not ?

If you have your own custom out-of-tree app, and you know that app will always be built with a specific toolchain / a toolchain supporting newlib, then you can also do:

```
# myapp/CMakeLists.txt

set(TOOLCHAIN_HAS_NEWLIB y)

find_package(Zephyr)

... rest of your CMake code ...
```

Note: you may even guard the `set()` to only take effect on certain toolchains, like this: `if(ZEPHYR_TOOLCHAIN_VARIANT STREQUAL cross-compile)`

Reply:
make sense, I will decline this PR, thanks.